dspace-server-webapp/src/test/java/org/dspace/app/rest/AuthenticationRestControllerIT.java

```diff
@@ -949,11 +976,29 @@ public void testShortLivedToken() throws Exception {
         getClient(token).perform(post("/api/authn/shortlivedtokens"))
             .andExpect(jsonPath("$.token", notNullValue()))
             .andExpect(jsonPath("$.type", is("shortlivedtoken")))
-            .andExpect(jsonPath("$._links.self.href", Matchers.containsString("/api/authn/shortlivedtokens")));
+            .andExpect(jsonPath("$._links.self.href", Matchers.containsString("/api/authn/shortlivedtokens")))
+            // Verify generating short-lived token doesn't change our CSRF token
+            // (so, neither the CSRF cookie nor header are sent back)
+            .andExpect(cookie().doesNotExist("DSPACE-XSRF-COOKIE"))
+            .andExpect(header().doesNotExist("DSPACE-XSRF-TOKEN"));
 
         assertEquals(salt, eperson.getSessionSalt());
     }
 
+    @Test
+    public void testShortLivedTokenWithCSRFSentViaParam() throws Exception {
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        // Same request as prior method, but this time we are sending the CSRF token as a querystring param.
+        // NOTE: getClient() method defaults to sending CSRF tokens as Headers, so we are overriding its behavior here
+        getClient(token).perform(post("/api/authn/shortlivedtokens").with(csrf()))
+            // BECAUSE we sent the CSRF token on querystring, it should be regenerated & a new token
+            // is sent back (in cookie and header).
+            .andExpect(cookie().exists("DSPACE-XSRF-COOKIE"))
+            .andExpect(header().exists("DSPACE-XSRF-TOKEN"));
```

Comment:
it would be better to verify that the token value is really changed

Reply:
@abollini : Unfortunately, I've not found a way in these tests to verify the "before" and "after" value of the CSRF token...the tests themselves generate the CSRF automatically by just calling `csrf()` (see line 994).  So, the reason I simply check the existence of the cookie & header is that those are ONLY returned in the response when the token changes.  If the token doesn't change, then neither the cookie nor the header exists (see the previous test on lines 982-983).